NuRadioReco/framework/event.py

```diff
@@ -59,8 +72,18 @@ def register_module_station(self, station_id, instance, name, kwargs):
         kwargs:
             the key word arguments of the run method
         """
-        if(station_id not in self.__modules_station):
+        if station_id not in self.__modules_station:
             self.__modules_station[station_id] = []
+            
+        # Not every kwargs argument passed to a run(..) method is serializable. Example is a detector object. 
+        # Drop all arguments for which this is the case
+        keys_to_be_dropped = [key for key in kwargs 
+                             if not is_serializable(kwargs[key])]
+        
+        for key in keys_to_be_dropped:
+            logger.warn(f"Dropping \"{key}\" of type {type(kwargs[key])} from kwargs of __modules_station")
+            kwargs.pop(key)
```

Comment:
I'm never 100% sure with Python, but kwargs is a dict, so my intuition would be that this is **not** a local copy and you're modifying it in-place, i.e. when you later pass kwargs back to the `run` method it will have been modified. Can you test this?

Reply:
damn, good point.

Reply:
I think running a deep copy is anyway adequate? (I did not test it though)